pkg/provider/handlers/secret_test.go

```diff
@@ -161,3 +163,69 @@ func TestSecretCreation(t *testing.T) {
 		})
 	}
 }
+
+func TestListSecrets(t *testing.T) {
+	mountPath, err := os.MkdirTemp("", "test_secret_creation")
+	if err != nil {
+		t.Fatalf("unexpected error while creating temp directory: %s", err)
+	}
+
+	defer os.RemoveAll(mountPath)
+
+	handler := MakeSecretHandler(nil, mountPath)
```

Comment:
You could create an interface named "NamespaceChecker" which has a single method. This would have a fake version for unit testing and a real version with a reference to the client.

Reply:
Another idea might be to try using the Store interface from containerd, and then only pass in the namespace store: https://github.com/openfaas/faasd/blob/acb5d0bd1cc5876d05261857b0717b779b95ec1b/vendor/github.com/containerd/containerd/namespaces/store.go#L29

Reply:
Thanks for the suggestion, have implemented the first approach